### PR TITLE
Feature/handle interrupted betterer

### PR DESCRIPTION
**Pull request by runningferret**
Handle this in the same way that we're issuing a PR against adhearsion/punchblock.

### CODE REVIEW

**Comment by runningferret** [comment]
@jared-prime @sfgeorge This is basically what we had discussed yesterday. Tested and behaves the same for our needs, and includes a unit test.

File: spec/punchblock/translator/asterisk/component/mrcp_prompt_spec.rb [hunk 1]
```diff
@@ -220,6 +220,18 @@ def expect_app_with_options(app, options)
                     end
                   end
 
+                  context "when the RECOG_STATUS variable is set to 'INTERRUPTED' after a successful recognition" do
+                    let(:recog_status) { 'INTERRUPTED' }
+                    let(:recog_completion_cause) { '000' }
```
Comment (sfgeorge) [inline spec/punchblock/translator/asterisk/component/mrcp_prompt_spec.rb:225]: Shall we mock completion_cause as empty here?

Comment (runningferret) [inline spec/punchblock/translator/asterisk/component/mrcp_prompt_spec.rb:225]: In the real world bug the completion cause is the last value it was, in this case '000' (success). I wanted to replicate the exact conditions we saw the issue.

**Comment by Jared-Prime** [comment]
:+1:

File: spec/punchblock/translator/asterisk/component/mrcp_prompt_spec.rb (hunk 1 above)
Comment (runningferret) [inline spec/punchblock/translator/asterisk/component/mrcp_prompt_spec.rb:225]: If we want to be REALLY pedantic we can also mock the RECOG_RESULT value to be the empty string, as that was the direct cause of the issue, but I don't know that that is necessarily useful

Comment (sfgeorge) [inline spec/punchblock/translator/asterisk/component/mrcp_prompt_spec.rb:225]: Cool, I agree with you then... good as is.

**Comment by sfgeorge** [comment]
Can you add a CHANGELOG entry for this fella?

**Comment by runningferret** [comment]
There is already a changelog entry for the other version of this indicating the same fix. Do you want another one?

**Comment by Jared-Prime** [comment]
@sfgeorge I shouldn't think a new CHANGELOG entry is required since this [commit is already merged](https://github.com/runningferret/punchblock/commit/9362f2d3755e6e3ab672722264e2755dbb5fe083)

**Comment by sfgeorge** [comment]
You guys are right!  Looks good.  Thank you! :+1: 

@ajayanty Can you merge this for us?

**Comment by ajayanty** [comment]
This has been merged into cloudvox:develop-ifbyphone

**Pull request merged**